### DISCUSSION TITLE
Relative import `../../../../../../*` is ugly. Why not make it default to for example `~/*` or `$/*` or `$lib/*` or `$components/*` etc on tsconfig.ts similar what SvelteKit doing?

### DISCUSSION THREAD

**Discussion by mashanz** [vercel/next.js#44268] [Ideas]
### Describe the feature you'd like to request

Import component like this is so ugly, not good DX.
```tsx
import SomeComponent from `../../../../../../../../components/TheComponent`
```

and its annoying to modify `tsconfig.ts` everytime creating new next project

### Describe the solution you'd like

I think we can make this config on `tsconfig.ts` as default
```ts
{
	"compilerOptions": {
		"paths": {
			"@/*": ["./*"]
		}
	}
}
```

so we can do more simple yet not uggly `../../../../` nonsense
```tsx
import SomeComponent from `@/components/TheComponent`
```

### Describe alternatives you've considered

I can make PR on this file:

[https://github.com/vercel/next.js/blob/canary/packages/create-next-app/templates/default/ts/tsconfig.json](https://github.com/vercel/next.js/blob/canary/packages/create-next-app/templates/default/ts/tsconfig.json)

but I think it might not the right solutions

**Comment by balazsorban44** [comment]
Hi, given this is a stylistic opinion, I don't think we can reasonably set a default (agree with the deeply nested paths, but still `@/` might not be that everyone would want.)

**Reply by joulev** [reply]
Agreed. I mistook such routes as NPM packages starting with `@` more than once. Now I use `~/` which is more readable for me personally.

**Reply by mashanz** [reply]
agree, just an example, anything works as long as it is not deep nested paths.

**Comment by mashanz** [comment]
May be we can use [$lib](https://kit.svelte.dev/docs/modules#$lib) or `$components` or other alternative?

**Comment by GabenGar** [comment]
The `@` prefix is wrong because it can clash with scoped npm packages. NodeJS introduced the official syntax for [internal import mapping](https://nodejs.org/api/packages.html#subpath-imports), so you have to change it in all your projects.

**Reply by mashanz** [reply]
ok, thats not my point, I change the discussion tittle for people not missing my point.

**Comment by mashanz** [comment]
Look like everyone is miss understanding what my poin is, so I change the title. 
my point is, why not using `$lib/*` or other alternative to replace `../../../../` like what SvelteKit default setup?
setup it manually to for example `~/*` or `$lib/*` every time creating a NextJS project is a pain and boring.

**Reply by GabenGar** [reply]
That's very opinionated and I think less of SvelteKit for doing that. If nextjs introduces such a rule then it will upset another subset of people (such as me), who'd have to change the default setting on each new project instead.

**Comment by ijjk** [comment]
Hi, the `@/* -> './*'` alias is now added by default in Next.js for JS and TS projects, give it a try with `pnpm create next-app@canary`

x-ref: https://github.com/vercel/next.js/pull/44672

**Reply by GabenGar** [reply]
Why not `#* -> './*'` though?

**Reply by ijjk** [reply]
`@/*` has more prior art and familiarity and we're reserving `#/*` for future usage with different semantics. Note you can still edit this after setting up a project and isn't required to be this way.
